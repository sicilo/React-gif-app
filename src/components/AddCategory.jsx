import React, { useState } from 'react'

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('Warcraft')

    const onInputChange = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();

        if (inputValue.trim().length <= 1) return;

        onNewCategory(inputValue.trim());
        setInputValue('');
    }

    return (
        <>
            <form onSubmit={onSubmit} aria-label='form'>
                <input type="text" placeholder='buscar gifs' value={inputValue} onChange={onInputChange} />
            </form>
        </>
    )
}
