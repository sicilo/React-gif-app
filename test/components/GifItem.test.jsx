import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas para <GifItem/>', () => {
    const title = "Gif";
    const url = "https://vitejs.dev/logo-with-shadow.png";

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe tener el titulo', () => {
        render(<GifItem title={title} url={url} />);

        expect(screen.getByText(title)).toBeTruthy();
    })

    test('Debe tener la imagen con la url y el alt indicado', () => {
        render(<GifItem title={title} url={url} />);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })
})