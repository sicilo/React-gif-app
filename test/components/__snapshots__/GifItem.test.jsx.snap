// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para <GifItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Gif"
      src="https://vitejs.dev/logo-with-shadow.png"
    />
    <p>
      Gif
    </p>
  </div>
</div>
`;
